### PR TITLE
feat: pull `create-dapp`, `xsnap`, and `agoric-cli` changes from release-getting-started branch

### LINKED ISSUE #8795: Pull `create-dapp` into master and official release
## What is the Problem Being Solved?

During hackweek we put together `@agoric/create-dapp` to enable `npm create`, and published it to NPM from a "fake" release branch. https://github.com/Agoric/agoric-sdk/tree/release-getting-started/packages/create-dapp

This is now relied upon by the new getting started guide, and should be pulled into master, and cherry picked in a real release branch.

### PR DESCRIPTION
**Pull request by LuqiPan**
closes: #8795 

## Description

This PR pulls the code for `create-dapp` so it can be merged into `master` branch

Note: I created this PR by first cherry-picking commits related to create-dapp in https://github.com/Agoric/agoric-sdk/compare/release-mainnet1B...release-getting-started then layering additional commits on top

Note2: I just cherry-picked all commits from https://github.com/Agoric/agoric-sdk/compare/release-mainnet1B...release-getting-started except https://github.com/Agoric/agoric-sdk/commit/d127d1db6a8d22a5c56c6408f69ae753c8215372 (merge commit) and https://github.com/Agoric/agoric-sdk/commit/a3805f72e974bdcd118b53a1e94425994472374c (change already in `master` branch) as requested by @michaelfig and @mhofman 

### Documentation Considerations

This PR updates `CHANGELOG.md` with relevant info.

### Testing Considerations

Ran sanity test and it was successful

PS: also open to other suggestions

### CODE REVIEW

File: packages/create-dapp/CHANGELOG.md [hunk 1]
```diff
@@ -2,3 +2,17 @@
 
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
+
+## 0.2.0 (2024-01-23)
+
+### Chores
+
+- Pull `create-dapp` into master branch and official release
+
```
Comment (mhofman) [inline packages/create-dapp/CHANGELOG.md]: You can leave this out, it will be populated by the `lerna` process

**Review by mhofman — COMMENTED** [review]
Please remove the "upcoming release" part of the Changelog changes.

If you want to use a conventional commit for the title of the PR, it likely should be `feat`, not `chore`.

That said I think we can keep it normal, and maybe avoid `master` in it (which will be confusing once cherry-picked in the release branch). Maybe:
> pull `create-dapp` from getting-started branch.

File: packages/xsnap/package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@agoric/xsnap",
-  "version": "0.14.2",
-  "description": "Description forthcoming.",
+  "version": "0.14.3-u12.0",
```
Comment (LuqiPan) [inline packages/xsnap/package.json]: @michaelfig and @mhofman should we bump the version here or leave it at `0.14.2`?

**Comment by LuqiPan** [comment]
Update: I just cherry-picked all commits from https://github.com/Agoric/agoric-sdk/compare/release-mainnet1B...release-getting-started except https://github.com/Agoric/agoric-sdk/commit/d127d1db6a8d22a5c56c6408f69ae753c8215372 (merge commit) and https://github.com/Agoric/agoric-sdk/commit/a3805f72e974bdcd118b53a1e94425994472374c (change already in `master` branch) as requested by @michaelfig and @mhofman

File: packages/xsnap/package.json (hunk 2 above)
Comment (mhofman) [reply packages/xsnap/package.json]: Oh right, please match any versions to what is currently on master

**Comment by mhofman** [comment]
@LuqiPan, there is no need to manually update "out of date" PRs unless there is conflicts, mergify will take case of it. But if you do, please don't use "merge", and instead "rebase" (our mergify check will currently allow these merge commits which we don't want)

File: packages/create-dapp/package.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "lint:types": "tsc -p jsconfig.json",
+    "lint:eslint": "eslint ."
+  },
+  "devDependencies": {
+    "ava": "^5.2.0",
+    "c8": "^7.13.0",
+    "dd-trace": "^3.3.0"
+  },
+  "dependencies": {
+    "agoric": "^0.21.2-u12.0"
```
Comment (mhofman) [inline packages/create-dapp/package.json]: Please change to the current version of `agoric-cli`

File: packages/xsnap/package.json (hunk 2 above)
Comment (mhofman) [reply packages/xsnap/package.json]: Please revert this change

**Review by mhofman — CHANGES_REQUESTED** [review]
Thanks for picking up the other commits. Please update the PR title to reflect.

For the requested changes, and previous review feedback, please interactive rebase and fixup the original commits so this PR can be cherry-picked into the release branch more cleanly, and remove the merge master commit.

File: packages/create-dapp/package.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    "test:xs": "exit 0",
+    "lint-fix": "yarn lint:eslint --fix",
+    "lint": "run-s --continue-on-error lint:*",
+    "lint:types": "tsc -p jsconfig.json",
+    "lint:eslint": "eslint ."
+  },
+  "devDependencies": {
+    "ava": "^5.2.0",
+    "c8": "^7.13.0",
+    "dd-trace": "^3.3.0"
```
Comment (mhofman) [inline packages/create-dapp/package.json]: I think we'll need to match these to currently used versions of deps in other packages.

Comment (LuqiPan) [reply packages/create-dapp/package.json]: Updated.

Actually on second thought, do we need `dd-trace` package here?

Comment (mhofman) [reply packages/create-dapp/package.json]: I believe we don't. Good call.

**Review by mhofman — APPROVED** [review]

File: packages/create-dapp/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@agoric/create-dapp",
-  "version": "0.1.0",
+  "version": "0.2.0",
```
Comment (mhofman) [inline packages/create-dapp/package.json]: Sorry I missed this change before approving. We do not manually bump versions in `agoric-sdk` and leave that to scripts when cutting a release. Please remove this change.

**Pull request merged**